src/transformers/modeling_gpt2.py

```diff
@@ -624,16 +623,35 @@ def forward(
             if output_hidden_states:
                 all_hidden_states = all_hidden_states + (hidden_states.view(*output_shape),)
 
-            outputs = block(
-                hidden_states,
-                layer_past=layer_past,
-                attention_mask=attention_mask,
-                head_mask=head_mask[i],
-                encoder_hidden_states=encoder_hidden_states,
-                encoder_attention_mask=encoder_attention_mask,
-                use_cache=use_cache,
-                output_attentions=output_attentions,
-            )
+            if getattr(self.config, "gradient_checkpointing", False):
```

Comment:
I think `if self.config.gradient_checkpointing:` is nicer

Reply:
Most model `config`s don't actually have this attribute, only the ones that support checkpointing (AFAIK, Bert and Longformer for now) so it's less risky to do things this way.

Reply:
This is in `modeling_gpt2.py` which only works with `configuration_gpt2.py`. So if you add `gradient_checkpointing` to the config with default = `False` I don't see why this would be risky